ui/pages/token-allowance/token-allowance.js

```diff
@@ -102,7 +105,11 @@ export default function TokenAllowance({
     return inputValue.replace(/,/gu, '.');
   };
 
-  let customPermissionAmount = replaceCommaToDot(customTokenAmount).toString();
+  let customPermissionAmount = TOKEN_ALLOWANCE_VALUE_REGEX.test(
```

Comment:
I know you didn't name this, but the regex variable name should explain what it evaluates instead of what it's used for.

Can you rename the constant with that in mind and include a brief comment above its definition:
```
// number with optional decimal point using a comma or dot
const NUM_W_OPT_DECIMAL_COMMA_OR_DOT_REGEX
```

That will make this code block more self-evident.

Reply:
@pedronfigueiredo You are right! I renamed the regex variable name and added a brief comment above its definition [here](https://github.com/MetaMask/metamask-extension/pull/17117/commits/4dd0c96f1b2a2fb259d51e90778ca95618544201). Thank you for your suggestion about this!